pkg/paioavail/paio_avail_listener.go

```diff
@@ -63,13 +65,14 @@ func NewAvailListener(availFromBlock uint64, repository *cRepos.InputRepository,
 		l1ReadDelay = aux
 	}
 	return AvailListener{
-		AvailFromBlock:     availFromBlock,
-		InputRepository:    repository,
-		InputReaderWorker:  w,
-		PaioDecoder:        paioDecoder,
-		L1CurrentBlock:     fromBlock,
-		ApplicationAddress: common.HexToAddress(applicationAddress),
-		L1ReadDelay:        l1ReadDelay,
+		AvailFromBlock:    availFromBlock,
+		AppRepository:     appRepository,
+		EpochRepository:   epochRepository,
```

Comment:
here I think we should abstract it from the reader.
So, to do this we could add a InputService in the base reader to do this kind of logic.